apps/admin/frontend/src/api.ts

```diff
@@ -102,3 +104,243 @@ export const unprogramCard = {
+    return ['getCurrentElectionMetadata'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () =>
+      apiClient.getCurrentElectionMetadata()
+    );
+  },
+} as const;
+
+export const getCastVoteRecordFiles = {
+  queryKey(): QueryKey {
+    return ['getCastVoteRecordFiles'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () => apiClient.getCastVoteRecordFiles());
+  },
+} as const;
+
+export const getCastVoteRecords = {
+  queryKey(): QueryKey {
+    return ['getCastVoteRecords'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () => apiClient.getCastVoteRecords());
+  },
+} as const;
+
+export const getCastVoteRecordFileMode = {
+  queryKey(): QueryKey {
+    return ['getCastVoteRecordFileMode'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () =>
+      apiClient.getCastVoteRecordFileMode()
+    );
+  },
+} as const;
+
+type GetWriteInsProps = QueryProps<'getWriteIns'>;
+export const getWriteIns = {
+  queryKey(props?: GetWriteInsProps): QueryKey {
+    return props ? ['getWriteIns', props] : ['getWriteIns'];
+  },
+  useQuery(props?: GetWriteInsProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () => apiClient.getWriteIns(props));
+  },
+} as const;
+
+type GetWriteInSummaryProps = QueryProps<'getWriteInSummary'>;
+export const getWriteInSummary = {
+  queryKey(props?: GetWriteInSummaryProps): QueryKey {
+    return props ? ['getWriteInSummary', props] : ['getWriteInSummary'];
+  },
+  useQuery(props?: GetWriteInSummaryProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () =>
+      apiClient.getWriteInSummary(props)
+    );
+  },
+} as const;
+
+type GetWriteInAdjudicationTableProps =
+  QueryProps<'getWriteInAdjudicationTable'>;
+export const getWriteInAdjudicationTable = {
+  queryKey(props?: GetWriteInAdjudicationTableProps): QueryKey {
+    return props
+      ? ['getWriteInAdjudicationTable', props]
+      : ['getWriteInAdjudicationTable'];
+  },
+  useQuery(props: GetWriteInAdjudicationTableProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () =>
+      apiClient.getWriteInAdjudicationTable(props)
+    );
+  },
+} as const;
+
+type GetWriteInImageProps = QueryProps<'getWriteInImage'>;
+export const getWriteInImage = {
+  queryKey(props?: GetWriteInImageProps): QueryKey {
+    return props ? ['getWriteInImage', props] : ['getWriteInImage'];
+  },
+  useQuery(props: GetWriteInImageProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () =>
+      apiClient.getWriteInImage(props)
+    );
+  },
+} as const;
+
+// Grouped Invalidations
+
+function invalidateCastVoteRecordQueries(queryClient: QueryClient) {
+  void queryClient.invalidateQueries(getCastVoteRecordFileMode.queryKey());
```

Comment:
I think @kofi-q and I dug into this and decided that we should always `await` the call to `invalidateQueries`. That way the mutation doesn't complete until the invalidated data is refetched, so the user will see whatever UI we have for "mutation in progress" until the effects of the mutation are ready to be rendered